tests/chainerx_tests/unit_tests/routines_tests/test_linalg.py

```diff
@@ -227,3 +227,76 @@ def forward_xp(self, inputs, xp):
         a, b = inputs
         out = xp.linalg.solve(a, b)
         return out,
+
+
+@op_utils.op_test(['native:0', 'cuda:0'])
+@chainer.testing.parameterize(*(
+    chainer.testing.product({
+        'shape': [(1, 1), (3, 3), (6, 6)],
+        'in_dtypes': ['float32', 'float64']
+    })
+))
+class TestCholesky(NumpyLinalgOpTest):
+
+    def generate_inputs(self):
+        a = numpy.random.random(self.shape).astype(self.in_dtypes)
+        # Make random square matrix a symmetric positive semi-definite one
+        a = numpy.array(a.T.dot(a))
+        return a,
+
+    def forward_xp(self, inputs, xp):
+        a, = inputs
+
+        # Input has to be symmetrized for backward test to work
+        def symmetrize(A):
+            L = xp.tril(A)
+            return (L + L.T)/2.
+        a = symmetrize(a)
+
+        L = xp.linalg.cholesky(a)
+        return L,
+
+
+@op_utils.op_test(['native:0', 'cuda:0'])
+@chainer.testing.parameterize(*(
+    chainer.testing.product({
+        'shape': [(), (2, 3), (3, 2), (6, 6)],
+        'in_dtypes': ['float32', 'float64'],
+    })
+))
+class TestCholeskyFailing(NumpyLinalgOpTest):
```

Comment:
Let's create tests (that are not stochastic) for positive semi-definite inputs.

Reply:
Added a test with three different positive semidefinite matrices.